src/main/kotlin/org/jitsi/nlj/rtcp/KeyframeRequester.kt

```diff
@@ -31,39 +39,128 @@ import org.jitsi.rtp.rtcp.rtcpfb.payload_specific_fb.RtcpFbFirPacketBuilder
  * on what the client supports
  * 3) Aggregation.  This class will pace outgoing requests such that we don't spam the sender
  */
-class KeyframeRequester : ObserverNode("Keyframe Requester") {
+class KeyframeRequester : TransformerNode("Keyframe Requester") {
+
+    companion object {
+        private const val DEFAULT_WAIT_INTERVAL_MS = 100
+    }
+
+    var waitIntervalMs = DEFAULT_WAIT_INTERVAL_MS
+
+    override fun transform(packetInfo: PacketInfo): PacketInfo? {
+        val packet = packetInfo.packet
+        val pliOrFir = when {
+            packet is CompoundRtcpPacket -> {
+                packet.packets.first { it is RtcpFbPliPacket || it is RtcpFbFirPacket }
+            }
+            packet is RtcpFbFirPacket -> packet
+            packet is RtcpFbPliPacket -> packet
+            else -> null
+        }
```

Comment:
i just realized i think we can make this slightly better by returning early if this is null.  we can do this by doing:
```
val pliOrFir = when {
            packet is CompoundRtcpPacket -> {
                packet.packets.first { it is RtcpFbPliPacket || it is RtcpFbFirPacket }
            }
            packet is RtcpFbFirPacket -> packet
            packet is RtcpFbPliPacket -> packet
            else -> null
        } ?: return null
```

now we skip the logic below and `pliOrFir` will be a non-nullable type, which i think is a bit better.